File: server/src/main/java/de/pnp/manager/network/session/Session.java
package de.pnp.manager.network.session;

import de.pnp.manager.main.Utility;
import de.pnp.manager.network.interfaces.Client;

import java.util.ArrayList;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

public class Session implements ISession {

    protected String sessionID;
    protected String sessionName;
    protected ArrayList<Client> clients;

    public Session(String sessionID, String sessionName) {
        this.sessionID = sessionID;
        this.sessionName = sessionName;
        this.clients = new ArrayList<>();
    }

    public void addClient(Client client) {
        this.clients.add(client);
    }

    public void removeClient(Client client) {
        this.clients.remove(client);
    }

    @Override
    public String getSessionID() {
        return sessionID;
    }

    @Override
    public String getSessionName() {
        return sessionName;
    }

    @Override
    public int getMaxClients() {
        return -1;
    }

    @Override
    public boolean isPasswordProtected() {
        return false;
    }

    @Override
    public String getHost() {
        return "Dungeon Master";
    }

    @Override
    public String getInfo() {
        return "";
    }

    @Override
    public Collection<Client> getParticipatingClients() {
        return clients;
    }

    @Override
    public Map<String, Object> getConfig() {
        Map<String, Object> config = new HashMap<>();
        Utility.getConfig().getKeys().forEachRemaining(key -> config.put(key, Utility.getConfig().getProperty(key)));
        return config;
    }
}

File: app/src/main/java/de/pnp/manager/app/model/Session.kt
package de.pnp.manager.app.model

import de.pnp.manager.network.client.IClient
import de.pnp.manager.network.session.ISession
import java.util.*

class Session : ISession {

    private var sessionID: String = ""
    fun setSessionId(value : String) {
        sessionID = value
    }
    override fun getSessionID() = sessionID

    private var sessionName: String = ""
    fun setSessionName(value : String) {
        sessionName = value
    }
    override fun getSessionName() = sessionName

    private var maxClients: Int = 0
    fun setMaxClients(value : Int) {
        maxClients = value
    }
    override fun getMaxClients() = maxClients

    private var passwordProtected: Boolean = false
    fun setPasswordProtected(value : Boolean) {
        passwordProtected = value
    }
    override fun isPasswordProtected() = passwordProtected

    private var host: String = ""
    fun setHost(value : String) {
        host = value
    }
    override fun getHost() = host

    private var info: String = ""
    fun setInfo(value : String) {
        info = value
    }
    override fun getInfo() = info

    private var participatingClients: MutableCollection<out IClient> = Collections.emptyList()
    fun setParticipatingClients(value : MutableCollection<out IClient>) {
        participatingClients = value
    }
    override fun getParticipatingClients() = participatingClients

    private var config: MutableMap<String, Any> = Collections.emptyMap()
    fun setConfig(value : MutableMap<String, Any>) {
        config = value
    }
    override fun getConfig() = config
}